include/internal/catch_tostring.h

```diff
@@ -16,6 +16,18 @@
 #include "catch_compiler_capabilities.h"
 #include "catch_stream.h"
 
+#ifndef CATCH_CONFIG_STRING_VIEW
```

Comment:
This belongs to `catch_compiler_capabilities.h` and should work a bit differently (there are other similar macros there already, you can use them as a rough template).

Reply:
Work differently how?
Using `_MSCVC_LANG` is necessary to detect whether the user is compiling with e.g. /std:c++17 on msvc, and is going to be necessary at least until the next major release .
Or do you mean putting this logic into some `CATCH_INTERNAL_MSVC_CPP17_OR_GREATER` macros?

Reply:
Yeah, this is much better -- in our experience, the autodetection will always break down on some obscure platform, so feature toggles need to let the user override them as they see fit, which is why there is the `INTERNAL_*` and `NO_*` distinction in defines.

2 more things:
1) `CATCH_CONFIG_STRING_VIEW` should be `CATCH_CONFIG_CPP17_STRING_VIEW` (respectively `CATCH_INTERNAL_CONFIG_CPP17_STRING_VIEW`)
2) Is there an actual need for MSVC-specific C++ standard defines? (e.g. cannot it just toggle `CATCH_CPP14_OR_GREATER` and `CATCH_CPP17_OR_GREATER`?)

Reply:
`__cplusplus` is still `199711L` on MSVC, and `_MSVC_LANG` matches what standard you selected with compiler flags.
With recent versions, you can also use `/Zc:__cplusplus` to make `__cplusplus` match `_MSVC_LANG`, but this will stay disabled by default until at least the next *major* release of MSVC. See https://blogs.msdn.microsoft.com/vcblog/2018/04/09/msvc-now-correctly-reports-__cplusplus/.